types/proposal.go

```diff
@@ -89,11 +90,26 @@ func (p *Proposal) ValidateBasic() error {
 //
 // For more information on the meaning of 'timely', see the proposer-based timestamp specification:
 // https://github.com/tendermint/spec/tree/master/spec/consensus/proposer-based-timestamp
-func (p *Proposal) IsTimely(recvTime time.Time, sp SynchronyParams) bool {
+func (p *Proposal) IsTimely(recvTime time.Time, sp SynchronyParams, round int32) bool {
+	// The message delay values are scaled as rounds progress.
+	// Every 10 rounds, the message delay is doubled to allow consensus to
+	// proceed in the case that the chosen value was too small for the given network conditions.
+	// For more information and discussion on this mechanism, see the relevant github issue:
+	// https://github.com/tendermint/spec/issues/371
+	maxShift := bits.LeadingZeros64(uint64(sp.MessageDelay)) - 1
```

Comment:
This is a closed PR, but I feel that this treatment should be restricted to when the shift is applied, namely `round > 10`, Don't you think that enclosing this code inside an if would be a better solution?